### PR TITLE
Daily Update Checker (on start up)

### PR DESCRIPTION
**Pull request by pljones**
Whilst I was doing the update checker, I couldn't stop myself making a few other small changes:
* Mention use of "-e localhost"
* Move saving settings to CSettings
* Move defaulting of centralserver into CSettings
* Server translations might be used even headless (logging)

The bulk of the changes is in the one commit.  It relies on the change to make CSettings responsible for saving and I used `tr()` liberally, even headless, so loading the translations might be useful.  The other two were very much "in passing" - the first triggered by some comment on Facebook, the third following my other changes (should have been done there).  Anyway...
* Add an update checker

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -755,11 +792,13 @@ QString UsageArguments ( char **argv )
         "  -n, --nogui           disable GUI\n"
         "  -p, --port            set your local port number\n"
         "  -t, --notranslation   disable translation (use englisch language)\n"
+        "      --updatetool      'daily' or 'never' check for a new " + APP_NAME + " version\n"
         "  -v, --version         output version information and exit\n"
         "\nServer only:\n"
         "  -a, --servername      server name, required for HTML status\n"
         "  -d, --discononquit    disconnect all clients on quit\n"
         "  -e, --centralserver   address of the central server\n"
+        "                        (or 'localhost' to be a central server)\n"
```
Comment (corrados) [inline src/main.cpp:801]: Good idea. I have just applied this to master, thanks.

File: src/settings.h [hunk 2]
```diff
@@ -36,25 +36,34 @@
...
+    Q_OBJECT
+
 public:
     CSettings() :
         vecWindowPosMain ( ), // empty array
         strLanguage      ( "" ),
-        strFileName      ( "" ) {}
+        strFileName      ( "" )
+    {
+        QObject::connect ( QCoreApplication::instance(), &QCoreApplication::aboutToQuit,
```
Comment (corrados) [inline src/settings.h:49]: That's a good idea. That improves the code. I've just added this commit to the Git master, thanks.

File: src/main.cpp [hunk 3]
```diff
@@ -533,13 +533,6 @@ int main ( int argc, char** argv )
 
 
     // Dependencies ------------------------------------------------------------
-    // per definition: if we are in "GUI" server mode and no central server
```
Comment (corrados) [inline src/main.cpp:536]: We need this here in case of a headless server since no settings are used in that case.

Comment (pljones) [reply src/main.cpp:536]: The inifile is used if specified and settings are _always_ loaded.  If no GUI is given settings are no _written_ to the inifile.  So the load is always called.

Comment (pljones) [reply src/main.cpp:550]: Rebase moved this to commit 2267a4d .

**Comment by pljones** [comment]
I'm going to close this off.

**Pull request closed without merging**